### DISCUSSION TITLE
Multi site Dragonfly in a Kubernetes environment

### DISCUSSION THREAD

**Discussion by JWThorne** [dragonflydb/dragonfly#2269] [Q&A]
I need highly available, multisite Redis or Redis equivalent key value store. Is there a way to deploy a dragonfly with high availability and high throughput in multiple sites?

Here's the problem. I need short term, ephemeral transactional data to be **_eventually_** consistent in two sites that are 4500 km apart, so we would want apps in the first location to only talk to the local instances. My RTO/RPO would need to be measured in seconds or minutes at most. We don't really need persistence, per se. I have VPN connectivity between the two sites. Its sensitive data so we currently aren't allowed to put it in a public cloud. 

As we have a kubernetes based system, we have to have it so the local sites have their own redundancy so we can put nodes into maintenance, etc. We don't use vmware or other virtualization, we are strictly bare metal kubernetes.

**Comment by chakaz** [comment]
Hi @JWThorne 
I'm afraid we do not currently support cross-site replication.
If O(minutes) delay is ok for you, I guess you could try to setup auto save and load, but that's a complicated setup with downtime for the replica, so it probably won't be a good fit.

May I ask for more details about your setup? How do you (plan to?) use Redis / Dragonfly, and what kind of load to you anticipate?

**Reply by JWThorne** [reply]
Thanks for the reply. We already have redis, using the bitnami chart for Cluster. That doesn't provide multi-site but it does provide local HA. I could replicate those volumes as well, but as you say, it's complicated. 

I'd be happy with a few minutes delay, but I'm trying to get down to sub 30 seconds. 

Data size is under 1 Gbyte right now, but I'm working on other projects that might increase it.  Request rate is about 1000 reads/sec, 300 writes/sec.

**Comment by jeremych1000** [comment]
@JWThorne did you ever find a suitable product for this? We are in the same boat and have investigated several solutions, e.g. the opstree redis operator is not bulletproof with failover.
